### PR TITLE
feat: rust client

### PR DESCRIPTION
**Pull request by BelfordZ**
depends on:
https://github.com/open-rpc/meta-schema/pull/84
https://github.com/open-rpc/examples/pull/85
https://github.com/open-rpc/spec/pull/144

### CODE REVIEW

File: templates/rs/static/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[package]
+name = "template-client"
+version = "1.0.0"
+authors = ["Zachary Belford <belfordz66@gmail.com>"]
+
+[dependencies]
```
Comment (shanejonas) [inline templates/rs/static/Cargo.toml]: this should include `jsonrpc_client_core` with the version number

File: src/getTypings.ts [hunk 2]
```diff
@@ -2,6 +2,16 @@ import { compile } from "json-schema-to-typescript";
 import { types } from "@open-rpc/meta-schema";
 import { generateMethodParamId, generateMethodResultId } from "@open-rpc/schema-utils-js";
 import { map, chain, startCase } from "lodash";
+const promiseSeries = require("promise-series");
```
Comment (BelfordZ) [inline src/getTypings.ts]: remove this

File: package.json [hunk 3]
```diff
@@ -35,8 +35,13 @@
     "is-url": "^1.2.4",
     "jayson": "^2.1.2",
     "jest": "^24.5.0",
+    "json-schema-faker": "^0.5.0-rc16",
```
Comment (BelfordZ) [inline package.json]: remove this - I was gonna use this to try to use a different tool than quicktype that would generate types from json examples, not json schema.

File: package.json [hunk 4]
```diff
@@ -35,8 +35,13 @@
...
     "json-schema-to-typescript": "^6.1.3",
     "lodash": "^4.17.11",
+    "promise-series": "^1.0.0",
```
Comment (BelfordZ) [inline package.json]: remove this

File: package.json [hunk 5]
```diff
@@ -35,8 +35,13 @@
...
+    "quicktype": "13.0.10",
+    "quicktype-core": "^6.0.26",
+    "transform-json-types": "^0.7.0",
```
Comment (BelfordZ) [inline package.json]: remove this

File: src/bootstrapGeneratedPackage.ts [hunk 6]
```diff
@@ -1,14 +1,31 @@
...
+export default (exec: any): TBootstrapGeneratedPackage => (destinationDirectoryName: string, language: string) => {
+
+  const commandSequence = [
+    `cd ${destinationDirectoryName}`,
+    ...languageCommandSequences[language],
   ].join(" && ");
 
-  return new Promise((resolve, reject) => exec(commandSequence, (err: Error, stdout: string) => {
+  return new Promise((resolve, reject) => exec(commandSequence, (err: Error, stdout: string, stdErr: string) => {
+    console.log(stdout, stdErr);
```
Comment (BelfordZ) [inline src/bootstrapGeneratedPackage.ts]: cargo seems to actually use stdErr, need to make necessary provisions if this is the case.

**Comment by BelfordZ** [comment]
closed in favor of https://github.com/open-rpc/generator-client/pull/108

**Pull request closed without merging**